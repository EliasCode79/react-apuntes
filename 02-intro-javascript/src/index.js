//  Operador ternario

let msj = 'Este es un msj';

let esVerdad = true

let msjSend = ( esVerdad && es)