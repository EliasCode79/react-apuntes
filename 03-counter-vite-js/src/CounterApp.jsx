import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {
	const handler = (event) => {
		// console.log(event)
		console.log('+1');
		value = 100;
	};

	return (
		<>
			<h1>Counter App</h1>
			<h2>{value}</h2>

			{/* onclick envia por defecto un event, en react podemos llamar a una funcion si (), cuando se envia un parametro */}
			<button onClick={handler}>+1</button>
		</>
	);
};

CounterApp.propTypes = {
	value: PropTypes.number.isRequired,
};

CounterApp.defaultPros = {
	value: 'No hay numero',
};
