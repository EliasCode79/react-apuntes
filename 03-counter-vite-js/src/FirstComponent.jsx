export const FirtsApp = () => {
	return <h2>Elias Franco</h2>;
};
