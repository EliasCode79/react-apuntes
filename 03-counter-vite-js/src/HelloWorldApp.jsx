// no estamos importando React, porque ya los estamos importando de manera global con Vite
export const App = () => {
	return <h1>Hola mundo</h1>;
};
