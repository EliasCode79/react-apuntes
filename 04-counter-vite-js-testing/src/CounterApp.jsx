import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {
	const [counter, setCounter] = useState(value);

	const handler = (operation) => {
		setCounter(counter + 1);
	};

	const handlerRest = () => setCounter(counter - 1);
	const handlerReset = () => setCounter(value);

	return (
		<>
			<h1>Counter App</h1>
			<h2>{counter}</h2>

			<button onClick={handler}>+1</button>
			<button onClick={handlerRest}>-1</button>
			<button aria-label='btn-reset' onClick={handlerReset}>
				Reset
			</button>
		</>
	);
};

CounterApp.propTypes = {
	value: PropTypes.number.isRequired,
};

CounterApp.defaultPros = {
	value: 'No hay numero',
};
