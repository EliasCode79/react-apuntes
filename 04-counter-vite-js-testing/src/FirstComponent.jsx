import PropTypes from 'prop-types';

export const FirtsApp = ({ title, subTitle, name }) => {
	return (
		<>
			<h1 data-testid='test-title'>{title}</h1>
			<p>{subTitle}</p>
			<p>{name}</p>
		</>
	);
};

// indicamos q tipos de datos esperamos y si es obligatorio o no
FirtsApp.propTypes = {
	title: PropTypes.string.isRequired, // indicamos q es obligatorio q el required.
	subTitle: PropTypes.string,
};

FirtsApp.defaultProps = {
	name: 'no hay nombre',
	// title: 'no hay titulo',
	subTitle: 'no hay subtitulo',
};
