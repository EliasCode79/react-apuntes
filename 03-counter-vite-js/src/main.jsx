import React from 'react';
import ReactDOM from 'react-dom/client';
import { FirtsApp } from './FirstComponent';
import { CounterApp } from './CounterApp';
// import { App } from './HelloWorldApp';
import './styles.css';

ReactDOM.createRoot(document.querySelector('#root')).render(
	<React.StrictMode>
		<CounterApp value={20} />
	</React.StrictMode>
);
