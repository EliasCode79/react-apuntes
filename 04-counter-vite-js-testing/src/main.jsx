import React from 'react';
import ReactDOM from 'react-dom/client';
import { FirtsApp } from './FirstComponent';
import { CounterApp } from './CounterApp';
// import { App } from './HelloWorldApp';
import './styles.css';

ReactDOM.createRoot(document.querySelector('#root')).render(
	<React.StrictMode>
		<FirtsApp title='HOla soy elias' />
		{/* <CounterApp value={20} /> */}
	</React.StrictMode>
);
