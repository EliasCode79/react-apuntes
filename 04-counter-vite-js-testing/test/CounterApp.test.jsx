import { render, screen } from '@testing-library/react';
import { CounterApp } from '../src/CounterApp';

describe('pruebas en el <CounterApp />', () => {
	const valorInicial = 100;
	test('Snaptshot del CounterApp', () => {
		const { container } = render(<CounterApp value={valorInicial} />);

		expect(container).toMatchSnapshot();
	});

	test('Debe mostrar el valor inical 100', () => {
		render(<CounterApp value={100} />);
		expect(screen.getByText('100')).toBeTruthy();
	});
});
