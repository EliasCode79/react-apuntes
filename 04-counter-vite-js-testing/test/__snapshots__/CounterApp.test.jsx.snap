// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el <CounterApp /> Snaptshot del CounterApp 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
    100
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button>
    Reset
  </button>
</div>
`;
